cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -37,12 +37,31 @@ void reader::impl::decode_page_data(size_t skip_rows, size_t num_rows)
       return cursum + _metadata->get_output_nesting_depth(chunk.src_col_schema);
     });
 
+  // Check to see if there are any string columns present. If so, then we need to get size info
+  // for each string page. This size info will be used to pre-allocate memory for the column,
+  // allowing the page decoder to write string data directly to the column buffer, rather than
+  // doing a gather operation later on.
+  // TODO: This step is somewhat redundant if size info has already been calculated (nested schema,
+  // chunked reader).
+  auto is_string_col = [](gpu::ColumnChunkDesc const& chunk) {
```

Comment:
This function is duplicated in `page_decode.cuh` and seems easy to get wrong and bug fix only one. Is there a way we can combine these?

Reply:
Should be able to do an inline or constexpr shared version. I'll give it a go.